uint16 digital_input
uint16[4] analog_input